src/main/policy.cc

```diff
@@ -317,3 +317,47 @@ int scanpolicy_from_jsobject(as_policy_scan* policy, Local<Object> obj, const Lo
 	as_v8_detail( log, "Parsing scan policy: success");
 	return AS_NODE_PARAM_OK;
 }
+
+int partitions_from_jsobject(as_partition_filter* pf,  bool* defined, v8::Local<v8::Object> obj, const LogInfo* log)
+{
+	int rc = AS_NODE_PARAM_OK;
+
+	*defined = false;
+	if ((rc = get_optional_bool_property(defined, NULL, obj, "pfEnabled", log)) != AS_NODE_PARAM_OK) {
+		return rc;
+	}
+
+	if(!*defined) {
+		return AS_NODE_PARAM_OK;
+	}
+	
+	*defined = true;
+	if (Nan::Has(obj, Nan::New("partFilter").ToLocalChecked()).FromJust()) {
+		Local<Value> pf_obj = Nan::Get(obj, Nan::New("partFilter").ToLocalChecked()).ToLocalChecked();
+		if (!pf_obj->IsUndefined() && pf_obj->IsObject()) {
+			if ((rc = get_optional_uint32_property((uint32_t*) &pf->begin, NULL, pf_obj.As<Object>(), "begin", log)) != AS_NODE_PARAM_OK) {
+				return rc;
+			}
+			if ((rc = get_optional_uint32_property((uint32_t*) &pf->count, NULL, pf_obj.As<Object>(), "count", log)) != AS_NODE_PARAM_OK) {
+				return rc;
+			}
+			int len = AS_DIGEST_VALUE_SIZE;
+			bool digest_defined = false;
+			if ((rc = get_optional_bytes_property((uint8_t **)&pf->digest.value, &len, &digest_defined, pf_obj.As<Object>(), "digest", log)) != AS_NODE_PARAM_OK) {
```

Comment:
Should `digest` be a String or a Buffer? The docs for the `Scan#partitions` function specify the type of the `digest` parameter as `string`:

     @param {string} digest